src/contracts/AbstractARM.sol

```diff
@@ -0,0 +1,689 @@
+    /// From a User perspective, this is the token being bought.
+    /// token1 is also compatible with the Uniswap V2 Router interface.
+    IERC20 public immutable token1;
+
+    ////////////////////////////////////////////////////
+    ///             Storage Variables
+    ////////////////////////////////////////////////////
+
+    /**
+     * @notice For one `token0` from a Trader, how many `token1` does the pool send.
+     * For example, if `token0` is WETH and `token1` is stETH then
+     * `traderate0` is the WETH/stETH price.
+     * From a Trader's perspective, this is the stETH/WETH buy price.
+     * Rate is to 36 decimals (1e36).
+     */
+    uint256 public traderate0;
+    /**
+     * @notice For one `token1` from a Trader, how many `token0` does the pool send.
+     * For example, if `token0` is WETH and `token1` is stETH then
+     * `traderate1` is the stETH/WETH price.
+     * From a Trader's perspective, this is the stETH/WETH sell price.
+     * Rate is to 36 decimals (1e36).
+     */
+    uint256 public traderate1;
+
+    /// @notice cumulative total of all withdrawal requests included the ones that have already been claimed
+    uint128 public withdrawsQueued;
+    /// @notice total of all the withdrawal requests that have been claimed
+    uint128 public withdrawsClaimed;
+    /// @notice cumulative total of all the withdrawal requests that can be claimed including the ones already claimed
+    uint128 public withdrawsClaimable;
+    /// @notice index of the next withdrawal request starting at 0
+    uint128 public nextWithdrawalIndex;
+
+    struct WithdrawalRequest {
+        address withdrawer;
+        bool claimed;
+        // When the withdrawal can be claimed
+        uint40 claimTimestamp;
+        // Amount of assets to withdraw
+        uint128 assets;
+        // cumulative total of all withdrawal requests including this one.
+        // this request can be claimed when this queued amount is less than or equal to the queue's claimable amount.
+        uint128 queued;
+    }
+
+    /// @notice Mapping of withdrawal request indices to the user withdrawal request data
+    mapping(uint256 requestId => WithdrawalRequest) public withdrawalRequests;
+
+    /// @notice The account that can collect the performance fee
+    address public feeCollector;
+    /// @notice Performance fee that is collected by the feeCollector measured in basis points (1/100th of a percent)
+    /// 10,000 = 100% performance fee
+    /// 2,000 = 20% performance fee
+    /// 500 = 5% performance fee
+    uint16 public fee;
+    /// @notice The performance fees accrued but not collected.
+    /// This is removed from the total assets.
+    uint112 public feesAccrued;
+    /// @notice The total assets at the last time performance fees were calculated.
+    /// This can only go up so is a high watermark.
+    uint128 public lastTotalAssets;
+
+    address public liquidityProviderController;
```

Comment:
`liquidityProviderController` sounds like it runs everything. Maybe just `whitelist`?

Reply:
It also enforces a cap on the total assets so it does more than just whitelisting.
How about `capManager`?

Reply:
Fine with that.